### PR TITLE
nia/docs: add content for automation with TFC driver

### PR DESCRIPTION
**Pull request by findkim**
[/docs/nia/network-drivers](https://consul-af95eime2-hashicorp.vercel.app/docs/nia/network-drivers) is updated to include new driver information and a table to easily digest features for comparison.

### CODE REVIEW

File: website/content/docs/nia/network-drivers.mdx [hunk 1]
```diff
@@ -11,45 +11,141 @@ Network drivers are used by Consul-Terraform-Sync to execute and update network
...
 
 ### Understanding Terraform Automation
 
-On startup, Consul-Terraform-Sync downloads and installs [Terraform](https://www.terraform.io/downloads.html), prepares the local environment, and generates Terraform configuration files that make up the root module for each task. Terraform configuration and execution for each task is organized as separate [Terraform workspaces](https://www.terraform.io/docs/state/workspaces.html). The state files for tasks are independent of each other.
+Consul-Terraform-Sync automates Terraform execution with templated configuration to carry out infrastructure changes. The auto-generated configuration leverages input variables sourced from Consul and builds on top of reusable Terraform modules published and maintained by HashiCorp partners and the community. Consul-Terraform-Sync can also run your custom built modules that suite your team's specific network automation needs.
```
Comment (lornasong) [inline website/content/docs/nia/network-drivers.mdx:29]:
```suggestion
Consul-Terraform-Sync automates Terraform execution with templated configuration to carry out infrastructure changes. The auto-generated configuration leverages input variables sourced from Consul and builds on top of reusable Terraform modules published and maintained by HashiCorp partners and the community. Consul-Terraform-Sync can also run your custom built modules that suit your team's specific network automation needs.
```

File: website/content/docs/nia/network-drivers.mdx [hunk 2]
```diff
@@ -11,45 +11,141 @@ Network drivers are used by Consul-Terraform-Sync to execute and update network
...
 
-~> **Note:** Although Terraform state files for task workspaces are independent, this does not guarantee the infrastructure changes from concurrent task executions are independent. Ensure that modules across all tasks are not modifying the same resource objects or have overlapping changes that may result in race conditions during automation.
+The network driver for Consul-Terraform-Sync determines how the Terraform automation operates.
+
+<Tabs>
+<Tab heading="Terraform Driver">
```
Comment (lornasong) [inline website/content/docs/nia/network-drivers.mdx:34]: ooo nice formatting

File: website/content/docs/nia/network-drivers.mdx [hunk 3]
```diff
@@ -11,45 +11,141 @@ Network drivers are used by Consul-Terraform-Sync to execute and update network
...
+1. Creates or discovers Terraform Enterprise workspaces corresponding to the configured tasks.
+2. Prepares the local environment and generates Terraform configuration files that make up the root module for each task.
+3. Packages the generated files and uploads them as a configuration version for the task's workspace on Terraform Enterprise.
+
+Once all workspaces are set up, Consul-Terraform-Sync monitors the Consul catalog for service changes. When relevant changes are detected, the Terraform Cloud driver dynamically updates input variables for that task directly as [workspace variables](https://www.terraform.io/docs/cloud/workspaces/variables.html) using the Terraform Cloud API. The driver then queues a run on the workspace, with auto-apply enabled, to update your network infrastructure.
+
+~> **Note:** Although workspaces for tasks are executed in isolated environments, this does not guarantee the infrastructure changes from concurrent task executions are independent. Ensure that modules across all tasks are not modifying the same resource objects or have overlapping changes that may result in race conditions during automation.
+
+#### Remote Workspaces
```
Comment (lornasong) [inline website/content/docs/nia/network-drivers.mdx:102]: nothing to do with your change and no changes required - just noting an observation that links to headers on the non-default tab (Terraform Cloud Driver Tab) don't bring you to the sub-section

**Review by lornasong — APPROVED** [review]
looks good! i like the screenshots you included!

File: website/content/docs/nia/network-drivers.mdx (hunk 3 above)
Comment (findkim) [reply website/content/docs/nia/network-drivers.mdx:102]: 🙁 i noticed that too, I'll ask around to see what can be done about it

**Pull request merged**